B/bwamem2/build_tarballs.jl

```diff
@@ -0,0 +1,58 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "bwamem2"
+version = v"2.2.1"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/bwa-mem2/bwa-mem2/releases/download/v$(version)/Source_code_including_submodules.tar.gz", "9b001bdc7666ee3f14f3698b21673714d429af50438b894313b05bc4688b1f6d")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/bwa-mem2-*/ext/safestringlib/
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    ..
+cd ../../
+make -j${nproc}
+install -Dvm 755 "bwa-mem2${exeext}" "${bindir}/bwa-mem2${exeext}"
+install -Dvm 755 "bwa-mem2.avx${exeext}" "${bindir}/bwa-mem2.avx${exeext}"
+install -Dvm 755 "bwa-mem2.avx2${exeext}" "${bindir}/bwa-mem2.avx2${exeext}"
+install -Dvm 755 "bwa-mem2.avx512bw${exeext}" "${bindir}/bwa-mem2.avx512bw${exeext}"
+install -Dvm 755 "bwa-mem2.sse41${exeext}" "${bindir}/bwa-mem2.sse41${exeext}"
+install -Dvm 755 "bwa-mem2.sse42${exeext}" "${bindir}/bwa-mem2.sse42${exeext}"
+install -Dvm 755 "bwa-mem2${libext}" "${libdir}/bwa-mem2${libext}"
```

Comment:
Uhm, the [variable](https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables) for the extension of shared libraries is `dlext`, there's no `libext`

Reply:
Also, the library is called libbwa. Apparently the binaries do not depend on it. I never used the library, only the binaries. I added the correct name but then there are no bindings yet. I will try to add that and test it.

Reply:
Sorry for my ignorance, I'm not experienced with building C/C++. So there is a libbwa.a, which is a static library that does not export anything, as I just read. But since the binaries do not depend on it, I now just removed the corresponding line.

Reply:
Yeah, we can't use a static library from Julia, but you can convert it to a shared library, if you want to call it from Julia.  Let me know what you want to do.

Reply:
Since there is no documentation regarding libbwa, I'm not sure how useful this would be. But the same applies to the libbwa from BWA_jll and BurrowsWheelerAligner.jl uses it. I just checked the patch used in BWA_jll. I can try to add it here, it looks like the situation is similar enough that I can do this without knowing what I'm doing :)

Reply:
Or not. I don't even know if that patch does this conversion.

Reply:
For the record, I managed to compile the shared library with
```diff
diff --git a/B/bwamem2/build_tarballs.jl b/B/bwamem2/build_tarballs.jl
index 94256b688..8fc9bbf19 100644
--- a/B/bwamem2/build_tarballs.jl
+++ b/B/bwamem2/build_tarballs.jl
@@ -7,7 +7,9 @@ version = v"2.2.1"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/bwa-mem2/bwa-mem2/releases/download/v$(version)/Source_code_including_submodules.tar.gz", "9b001bdc7666ee3f14f3698b21673714d429af50438b894313b05bc4688b1f6d")
+    ArchiveSource("https://github.com/bwa-mem2/bwa-mem2/releases/download/v$(version)/Source_code_including_submodules.tar.gz",
+                  "9b001bdc7666ee3f14f3698b21673714d429af50438b894313b05bc4688b1f6d"),
+    DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across all platforms
@@ -19,6 +21,7 @@ cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
     -DCMAKE_BUILD_TYPE=Release \
     ..
 cd ../../../
+atomic_patch -p1 ../patches/position-independent-code.patch
 make -j${nproc}
 install -Dvm 755 "bwa-mem2${exeext}" "${bindir}/bwa-mem2${exeext}"
 install -Dvm 755 "bwa-mem2.avx${exeext}" "${bindir}/bwa-mem2.avx${exeext}"
@@ -26,6 +29,7 @@ install -Dvm 755 "bwa-mem2.avx2${exeext}" "${bindir}/bwa-mem2.avx2${exeext}"
 install -Dvm 755 "bwa-mem2.avx512bw${exeext}" "${bindir}/bwa-mem2.avx512bw${exeext}"
 install -Dvm 755 "bwa-mem2.sse41${exeext}" "${bindir}/bwa-mem2.sse41${exeext}"
 install -Dvm 755 "bwa-mem2.sse42${exeext}" "${bindir}/bwa-mem2.sse42${exeext}"
+cc -shared -fPIC -o ${libdir}/libbwa.${dlext} $(flagon -Wl,--whole-archive) libbwa.a $(flagon -Wl,--no-whole-archive) -lz
 """
 
 # These are the platforms we will build for by default, unless further
@@ -45,12 +49,14 @@ products = [
     ExecutableProduct("bwa-mem2.sse42", :bwamem2_sse42),
     ExecutableProduct("bwa-mem2.avx", :bwamem2_avx),
     ExecutableProduct("bwa-mem2.avx2", :bwamem2_avx2),
-    ExecutableProduct("bwa-mem2", :bwamem2)
+    ExecutableProduct("bwa-mem2", :bwamem2),
+    LibraryProduct("libbwa", :libbwa),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a"))
+    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a")),
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
diff --git a/B/bwamem2/bundled/patches/position-independent-code.patch b/B/bwamem2/bundled/patches/position-independent-code.patch
new file mode 100644
index 000000000..c7f0546b2
--- /dev/null
+++ b/B/bwamem2/bundled/patches/position-independent-code.patch
@@ -0,0 +1,13 @@
+diff --git a/Makefile b/Makefile
+index 359585f..ea59a91 100644
+--- a/Makefile
++++ b/Makefile
+@@ -90,7 +90,7 @@ else
+ myall:multi
+ endif
+ 
+-CXXFLAGS+=	-g -O3 -fpermissive $(ARCH_FLAGS) #-Wall ##-xSSE2
++CXXFLAGS+= -fPIC -g -O3 -fpermissive $(ARCH_FLAGS) #-Wall ##-xSSE2
+ 
+ .PHONY:all clean depend multi
+ .SUFFIXES:.cpp .o
```
but loading of the library actually fails because they rely on the main program defining a couple of global variables
https://github.com/bwa-mem2/bwa-mem2/blob/cf4306a47dac35e7e79a9e75398a35f33900cfd0/src/profiling.h#L34
https://github.com/bwa-mem2/bwa-mem2/blob/cf4306a47dac35e7e79a9e75398a35f33900cfd0/src/main.cpp#L40
which I don't think can possibly work in a sane way with a shared library.  I don't know why they thing this is a good idea.  Then I'm going to merge this PR as is.

Reply:
Cool, thanks a lot.